test/layer.js

```diff
@@ -17,66 +10,13 @@ test("should resolve layers of import statements", checkFixture, "layer")
 test(
   "should correctly wrap imported at rules in layers",
   checkFixture,
-  "layer-import-atrules",
-  {
-    nameLayer: hashLayerName,
-  }
+  "layer-import-atrules"
 )
 
 test(
   "should correctly wrap imported at rules in anonymous layers",
   checkFixture,
-  "layer-import-atrules-anonymous",
-  {
-    nameLayer: hashLayerName,
-  }
+  "layer-import-atrules-anonymous"
 )
 
-test("should group rules", checkFixture, "layer-rule-grouping", {
-  nameLayer: hashLayerName,
-})
-
-test("should pass the root file name to the nameLayer function", t => {
-  return postcss()
-    .use(atImport({ path: "test/fixtures/imports", nameLayer: hashLayerName }))
-    .process('@import "foo.css" layer;', { from: "layer.css" })
-    .then(result => {
-      t.is(result.css, `@layer import-anon-layer-52ff1597784c{\nfoo{}\n}`)
-    })
-})
-
-test("should error when value is not a function", t => {
-  return postcss()
-    .use(atImport({ nameLayer: "not a function" }))
-    .process("", { from: undefined })
-    .catch(error =>
-      t.regex(error.message, /nameLayer option must be a function/)
-    )
-})
-
-test("should throw when using anonymous layers without the nameLayer plugin option", t => {
-  return postcss()
-    .use(atImport({ path: "test/fixtures/imports" }))
-    .process('@import "foo.css" layer;', { from: undefined })
-    .catch(err => {
-      t.is(
-        err.message,
-        'postcss-import: <css input>:1:1: When using anonymous layers in @import you must also set the "nameLayer" plugin option'
-      )
-    })
-})
-
-function hashLayerName(index, rootFilename) {
-  if (!rootFilename) {
-    return `import-anon-layer-${index}`
-  }
-
-  // A stable, deterministic and unique layer name:
-  // - layer index
-  // - relative rootFilename to current working directory
-  return `import-anon-layer-${crypto
-    .createHash("sha256")
-    .update(`${index}-${rootFilename.split(cwd)[1]}`)
-    .digest("hex")
-    .slice(0, 12)}`
-}
+test("should group rules", checkFixture, "layer-rule-grouping")
```

Comment:
Is this description still accurate?

Also https://github.com/postcss/postcss-import/blob/3d51fe5009dbaccf7e4d1bdac75a1d7478c36f00/test/media.js#L28

Reply:
### layer

That entire test isn't that relevant anymore because the mechanic it was testing no longer exists.

If I remove it the test coverage also remains the same.

It does however contain a seemingly unrelated part.
This test has duplicate anonymous layered imports which merit a dedicated test.

I've added a test with and without `skipDuplicates` for duplicate anonymous layered imports and I've removed the `layer-rule-grouping` test.

--------

### media

I've relabeled this test from `media-join` to `media-combine`.

I think `join` is too easily associated with an array join into a string.

`combine` is more general and remains correct no matter the underlying method (nesting at rules vs. concatenating at rule params)